src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,659 @@
+import os
+import warnings
+import numpy as np
+import json
+import spikeinterface
+from spikeinterface.core.job_tools import fix_job_kwargs
+from spikeinterface.qualitymetrics import get_quality_metric_list, get_quality_pca_metric_list
+from spikeinterface.postprocessing import get_template_metric_names
+from pathlib import Path
+
+default_classifier_search_spaces = {
+    "RandomForestClassifier": {
+        "n_estimators": [100, 150],
+        "criterion": ["gini", "entropy"],
+        "min_samples_split": [2, 4],
+        "min_samples_leaf": [2, 4],
+        "class_weight": ["balanced", "balanced_subsample"],
+    },
+    "AdaBoostClassifier": {
+        "learning_rate": [1, 2],
+        "n_estimators": [50, 100],
+        "algorithm": ["SAMME", "SAMME.R"],
+    },
+    "GradientBoostingClassifier": {
+        "learning_rate": [0.05, 0.1],
+        "n_estimators": [100, 150],
+        "max_depth": [2, 4],
+        "min_samples_split": [2, 4],
+        "min_samples_leaf": [2, 4],
+    },
+    "SVC": {
+        "C": [0.001, 10.0],
+        "kernel": ["sigmoid", "rbf"],
+        "gamma": [0.001, 10.0],
+        "probability": [True],
+    },
+    "LogisticRegression": {
+        "C": [0.001, 10.0],
+        "solver": ["newton-cg", "lbfgs", "liblinear", "sag", "saga"],
+        "max_iter": [100, 500],
+    },
+    "XGBClassifier": {
+        "max_depth": [2, 4],
+        "eta": [0.2, 0.5],
+        "sampling_method": ["uniform"],
+        "grow_policy": ["depthwise", "lossguide"],
+    },
+    "CatBoostClassifier": {"depth": [2, 4], "learning_rate": [0.05, 0.15], "n_estimators": [100, 150]},
+    "LGBMClassifier": {"learning_rate": [0.05, 0.15], "n_estimators": [100, 150]},
+    "MLPClassifier": {
+        "activation": ["tanh", "relu"],
+        "solver": ["adam"],
+        "alpha": [1e-7, 1e-1],
+        "learning_rate": ["constant", "adaptive"],
+        "n_iter_no_change": [32],
+    },
+}
+
+
+class CurationModelTrainer:
+    """
+    Used to train and evaluate machine learning models for spike sorting curation.
+
+    Parameters
+    ----------
+    labels : list of lists, default: None
+        List of curated labels for each unit; must be in the same order as the metrics data.
+    output_folder : str, default: None
```

Comment:
```suggestion
    folder : str, default: None
```
We recently unified the API to use a straight `folder` instead of `output_folder`. Can you modify it in this new curation functions too?